libsolidity/formal/CHC.cpp

```diff
@@ -642,10 +642,20 @@ void CHC::externalFunctionCall(FunctionCall const& _funCall)
 	}
 
 	auto postCallState = vector<smtutil::Expression>{state().state()} + currentStateVariables();
-	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState);
+	auto error = errorFlag().increaseIndex();
+	auto nondet = (*m_nondetInterfaces.at(m_currentContract))(preCallState + postCallState + vector<smtutil::Expression>{error, state().thisAddress()});
 	// TODO this could instead add the summary of the called function, where that summary
 	// basically has the nondet interface of this summary as a constraint.
 	m_context.addAssertion(nondet);
+	solAssert(m_errorDest, "");
+	connectBlocks(m_currentBlock, predicate(*m_errorDest), errorFlag().currentValue() > 0);
+	// To capture the possibility of a reentrant call, we record in the call graph that the  current function
+	// can call any of the external methods of the current contract.
+	solAssert(m_currentContract && m_currentFunction, "");
+	for (auto const* base: m_currentContract->annotation().linearizedBaseContracts)
+		for (auto const* definedFunction: base->definedFunctions())
```

Comment:
We might actually have to do virtual resolution here (similar to `SMTEncoder`) otherwise it thinks it can run a function that it actually can't.